### PR TITLE
Bug 1172779 - [Customizer Launcher] displays "Communications" instead of "Dialer" and "Contacts"

### PR DESCRIPTION
**Pull request by punamdahiya**

### CODE REVIEW

File: app/js/controller/list_controller.js [hunk 1]
```diff
@@ -53,6 +53,12 @@ export default class ListController extends Controller {
   }
 
   launchApp(data) {
+    if (data.name == 'Phone') {
```
Comment (justindarc) [inline app/js/controller/list_controller.js]: I'm not sure about keying off the `name` here for launching via entry point. Perhaps we should just set an `entryPoint` property in the `data` object that gets passed in here that specifies an optional value. If `data.entryPoint` exists, then we use it.

File: app/js/model/list_model.js [hunk 2]
```diff
@@ -48,12 +48,45 @@ export default class ListModel extends Model {
     return detail;
   }
 
+  fillAppDetailsFromEntryPoints(app, appName) {
+    var detail = Object.create(null);
+    detail.manifestURL = app.manifestURL;
+    detail.description = app.manifest.description;
+    detail.author = app.manifest.developer ? app.manifest.developer.name : '';
+    detail.app = app;
+    if (appName == 'contacts') {
```
Comment (justindarc) [inline app/js/model/list_model.js]: Is there a way we can do this dynamically by iterating the `manifest.entry_points`?

File: app/js/controller/list_controller.js (hunk 1 above)
Comment (punamdahiya) [inline app/js/controller/list_controller.js]: Updated to launch using entry_point property added to detail

File: app/js/model/list_model.js (hunk 2 above)
Comment (punamdahiya) [inline app/js/model/list_model.js]: Updated patch to iterate manifest.entry_points

File: app/js/model/list_model.js [hunk 3]
```diff
@@ -48,12 +48,36 @@ export default class ListModel extends Model {
...
   getAppList(allApps) {
     return new Promise((resolve, reject) => {
       var installedApps = Object.create(null);
       var filterList = this.filterApps(allApps);
-      filterList.forEach(app =>
-        installedApps[app.manifestURL] = this.fillAppDetails(app));
+      filterList.forEach(app => {
+        // Check for Communications app, only app that uses
+        // multiple entry_points in manifest
+        if (app.manifestURL == 'app://communications.gaiamobile.org/manifest.webapp') {
```
Comment (justindarc) [inline app/js/model/list_model.js]: Can we just check for `app.manifest.entry_points` instead?

Comment (punamdahiya) [inline app/js/model/list_model.js]: +1, app.manifest.entry_points works and remove one more hard coded variable! will update patch. Thanks!

**Pull request merged**